tests/transaction.test.js

```diff
@@ -1,15 +1,22 @@
 const assert = require('assert');
-const { Transaction } = require('../src/blockchain');
-const { createSignedTx, signingKey } = require('./helpers');
```

Comment:
I get where you're coming from, this is kind of a debate in the developer community anyways: but I personally think having everything related to a test, inside the test file itself is really important. As of here, this is really a simple test it looks fine to just import some function to be used from an external source. But when the codebase grows (If have plans to add more stuff hopefully 🤞🏼), There will be more dependencies for tests that are imported from other places.

That may introduce a few more issues, mainly:
- Difficulty in readability: you might need to go back and forth in files to understand the tests.
- Difficulty to update: if we needed to change stuff from depending functions with a need for a specific test, that might be troublesome as they may have links with many other tests.

Again, at the current stage of the code, we don't need to think about all these but it's nice to have such standards in place before things get out of control. I encourage you to read [this](https://thoughtbot.com/blog/the-self-contained-test) article to get more sense of the idea of why is it ok to have duplicate code in tests for example.

Reply:
I get your point. However, the article does mention that it's not black and white. 

> There are some times when extracting setup is not only okay, but preferred. Your test might need a database to run. Or perhaps some factories to generate that data. Define these outside of your test.

In my opinion, generating a signed transaction is a bit like a factory generating dummy data. And since it's used by other tests, I would prefer to keep it separate.

But I do understand your point about readability. Yes, it means having to jump to a separate file. But in this case, the function is pretty self explanatory. Generate a correctly signed transaction. That code shouldn't live in more than 1 place.

Reply:
Hmm, sure. yeah currently, it doesn't seem that obvious, and yeah I agree it's in a gray area, I was just thinking of future-proofing, but yeah it's not that much to consider tbh. will revert it. 👍🏼